src/mongo/tree/MongoCollectionTreeItem.ts

```diff
@@ -101,46 +102,41 @@ export class MongoCollectionTreeItem implements IAzureParentTreeItem {
 	}
 
-	//tslint:disable:cyclomatic-complexity
-	executeCommand(name: string, args?: string[]): Thenable<string> {
+	executeCommand(name: string, args?: string[]): Thenable<string> | null {
+		//const requiresOneArg = ['findOne', 'insertMany', 'insertOne', 'insert', 'deleteOne', 'deleteMany', 'remove'];
+		const parameters = args ? args.map(parseJSContent) : undefined;
 		try {
-			if (name === 'findOne') {
-				return reportProgress(this.findOne(args ? args.map(parseJSContent) : undefined), 'Finding');
-			}
-			if (name === 'drop') {
-				return reportProgress(this.drop(), 'Dropping collection');
-			}
-			if (name === 'insertMany') {
-				return reportProgress(this.insertMany(args ? args.map(parseJSContent) : undefined), 'Inserting documents');
-			}
-			else {
-				let argument;
-				if (args && args.length > 1) {
-					return undefined;
-				}
-				if (args) {
-					argument = args[0];
-				}
-				if (name === 'insert') {
-					return reportProgress(this.insert(argument ? parseJSContent(argument) : undefined), 'Inserting document');
-				}
-				if (name === 'insertOne') {
-					return reportProgress(this.insertOne(argument ? parseJSContent(argument) : undefined), 'Inserting document');
-				}
-				if (name === 'deleteOne') {
-					return reportProgress(this.deleteOne(argument ? parseJSContent(argument) : undefined), 'Deleting document');
-				}
-				if (name === 'deleteMany') {
-					return reportProgress(this.deleteMany(argument ? parseJSContent(argument) : undefined), 'Deleting documents');
+			// tslint:disable-next-line:no-any
+			type MongoFunction = (...args: any[]) => Thenable<string>;
+			let functions: { [functionName: string]: [MongoFunction, string, number, number, number] } = {
+				// format: command name (from the argument) : corresponding function call[0],
+				// text to show during the operation[1], min #args[2], #max args[3], #args this function handles[4]
+				"drop": [this.drop, 'Dropping collection', 0, 0, 0],
+				"insert": [this.insert, 'Inserting document', 1, 1, 1],
+				"count": [this.count, 'Counting documents', 0, 2, 2],
+				"findOne": [this.findOne, 'Finding Document', 0, 2, 2],
+				"insertMany": [this.insertMany, 'Inserting documents', 1, 2, 2],
+				"insertOne": [this.insertOne, 'Inserting document', 1, 2, 2],
+				"deleteOne": [this.deleteOne, 'Deleting document', 1, 2, 1],
+				"deleteMany": [this.deleteMany, 'Deleting documents', 1, 2, 1],
+				"remove": [this.remove, 'Deleting document(s)', 1, 2, 1]
+			};
+
+			if (name in functions) {
+				let functionData = functions[name];
+				if (parameters.length < functionData[2]) { //has less than the min allowed
+					return Promise.reject(new Error(`Too few arguments passed to command ${name}.`));
 				}
-				if (name === 'remove') {
-					return reportProgress(this.remove(argument ? parseJSContent(argument) : undefined), 'Removing');
+				if (parameters.length > functionData[3]) { //has more than the max allowed
```

Comment:
these types of things are always easier to read (and self-documenting) if at the beginning of the block you do this:
```
  let funcCall = functionData[0];
  let whatever = functionData[1];
  let whatever = functionData[2];
  let maxParams = functionData[3]
```
etc